src/app/dataValidator/helpers/validators/validateType.js

```diff
@@ -1,13 +1,31 @@
 const { log, throwError } = require('../../utilities/messaging');
 
+const validateTypeOf = (schemaType, dataNode, parentSchemaName) => {
+  if (schemaType === `${typeof dataNode}`) {
+    log(`- Valid type of ${typeof dataNode}`);
+  } else {
+    throwError(
+      `Error: Type does not match for '${parentSchemaName}' - expected '${schemaType}' got '${typeof dataNode}'`,
+    );
+  }
+};
+
+const validateArray = (dataNode, parentSchemaName) => {
+  if (Array.isArray(dataNode)) {
+    log('- Valid type of array');
+  } else {
+    throwError(
+      `Error: Type does not match for '${parentSchemaName}' - expected 'array'`,
+    );
+  }
+};
+
 module.exports.validateType = (schemaType, dataNode, parentSchemaName) => {
   if (dataNode !== null) {
-    if (schemaType === `${typeof dataNode}`) {
-      log(`- Valid type of ${typeof dataNode}`);
+    if (schemaType === 'array') {
+      validateArray(dataNode, parentSchemaName);
```

Comment:
Actually, not sure this is worth splitting out like this - we're basically repeating the `if (CHECK) log(success) else error(message)` logic in the two subfunctions, when the only thing that changes is the `CHECK`.

I'd propose:

```js
const valid = isValidTypeOf(schemaType, dataNode, parentSchemaName); // THIS function handles the array vs non-array logic
if (valid) {
    log
}
else {
    throwError
}
```

Should make the unit tests nicer too, as you can assert a boolean response from `isValidTypeOf` rather than try to catch errors or test the absence of them.

Reply:
This is a nice idea, however I think I'd go for `isValidType`. Part of the reason I split the functions was to be explicit that `typeof` in JS does not match up with `type` in OpenAPI 3.0, due to the fact `typeof [] === 'object'`

Reply:
So just rename `isValidTypeOf` to `isValidType`? Yeah that seems fine to me 👍